app/src/ui/diff/side-by-side-diff.tsx

```diff
@@ -114,10 +114,29 @@ interface ISideBySideDiffProps {
   readonly showSideBySideDiff: boolean
 }
 
-interface ISideBySideDiffState {
+interface ISideBySideTextDiffDocumentContents {
+  /** The content lines of the "new" file */
+  readonly newContentLines: ReadonlyArray<string>
+
+  /** The content lines of the "old" file */
+  readonly oldContentLines: ReadonlyArray<string>
+
+  /** Whether or not the file can be expanded */
+  readonly canBeExpanded: boolean
+}
+
+interface ISideBySideTextDiffDocument {
   /** The diff that should be rendered */
   readonly diff: ITextDiff
 
+  /** Contents of the old and new files for this diff. */
+  readonly contents: ISideBySideTextDiffDocumentContents | null
+}
+
+interface ISideBySideDiffState {
+  /** Represents a diff and its associated files (old and new) */
+  readonly document: ISideBySideTextDiffDocument
```

Comment:
Not a fan of the `document` word, since CodeMirror also uses it, but I couldn't come up with a better name 🤣

Reply:
Would diffDocument work just to give it that differentiation?

Reply:
That feels a bit redundant, but definitely better! In any case, after some chat with @niik about how to improve diff rendering performance, I might get rid of this structure entirely 😅